exercises/practice/leap/.approaches/introduction.md

```diff
@@ -0,0 +1,96 @@
+# Introduction
+
+Every fourth year is a leap year (with some exceptions), but let's consider this one condition first. 
+
+To solve the Leap problem, we must determine if a year is evenly divisible by a number or if a reminder of an integer division is zero. 
+Such operation in computing is called [modulo][modulo]. 
+
+Unlike many languages, Elixir does not have [operators][operators] for either integer division or modulo. 
+Instead, it provides [`rem/2`][rem] guard and the [`Integer.mod/2`][mod] function.
+
+The two functions differ in how they work with negative numbers, but since, in this exercise,
+all the numbers are non-negative, both could work, depending on the approach.
+
+## General Solution
+
+To check if a year is divisible by `n`, we can do `rem(year, n) == 0`. We can define a function to make the intent clearer. 
+
+```elixir
+defp divides?(number, divisor), do: rem(number, divisor) == 0 
+```
+
+Any approach to the problem will perform this check three times to check if a year is equally divisible by 4, 100 and 400.
+What will differ is what Elixir features we will use to combine the checks. 
+
+## Approach: Boolean Operators
+
+The full rules are as follows:
+A year is a leap year if 
+* it is divisible by 4 
+* but not divisible by 100
+* unless it is divisible by 400
+
+We can use [boolean operators][boolean-operators] to combine the checks, for example, like so: 
+
+```elixir
+divides?(year, 400) or (not(divides?(year, 100))) and divides?(year, 4)
+```
+In the [boolean operators appraoch][operators-approach] we discuss the details of the solution.
+It includes variations of the operators and their precendence.
+
+## Approach: Function Guards
+
+Instead of using boolean operators, we can define multiple `leap_year?/1` functions with different guards. 
+We can use the order of the definitions to ensure correct check.
```

Comment:
This sounds wrong somehow, but I can't quite define it... "we can (...)" sounds like it's optional? But it's not optional to put the clauses in the right order, the order always matters. Maybe we can remove this sentence from here
```suggestion
```

and then in the detailed approach description mention something like:
````
Remember that the order of function clauses matters, and only the first clause that matched a specific function call will get executed. All the other clauses will be ignored. This means that we **don't** need to repeat out checks:
```elixir
# Don't copy this!
def leap_year?(year) when rem(year, 400) == 0, do: true
def leap_year?(year) when rem(year, 100) == 0 and not rem(year, 400) == 0, do: false
def leap_year?(year) when rem(year, 4) == 0 and not rem(year, 100) == 0 and not rem(year, 400) == 0, do: true
def leap_year?(year) when not rem(year, 4) == 0 and not rem(year, 100) == 0 and not rem(year, 400) == 0,, do: false
```
````